### ISSUE TITLE
Upgrade to Google's new Material Design Library

### ISSUE THREAD

**Issue by xcffl** [wallabag/android-app#809]
As described [here](https://medium.com/@neerajmoudgil/upgrading-to-new-android-material-design-components-e62ddb03c3d2), Google has moved to the new material design library for a period, which is much prettier. But wallabag still uses the legency one.

It's easy to do the migration. Simply doing some text replacement as described in the article above makes it. If necessary, I can do a pr.

**Comment by di72nn** [comment]
Migration to AndroidX is being done in #803 (not merged yet), themes are not changed though.

**Comment by xcffl** [comment]
Yes, it's because we're not using the `@style/ThemeOverlay.MaterialComponents` style but a customized one. I'll going to investigating how to integrate the new style into Wallabag. 

After that, I'll try to make Wallabag mimic the [Fortnightly](https://material.io/design/material-studies/fortnightly.html#about-fortnightly) news app example showed in the Material Design guideline, which I personally believe is a good example for text-centric apps.

![Fortnightly](https://storage.googleapis.com/spec-host/mio-staging%2Fmio-design%2F1563837804615%2Fassets%2F1XioSfaIy6D-f89WebLO_hL3C5e_IOXMn%2Fcasestudies-fortnightly-brandattributes.png)

I guess inserting [values](https://github.com/flutter/flutter/blob/master/examples/flutter_gallery/lib/demo/fortnightly/fortnightly.dart) with Material Theming will do.

**Comment by xcffl** [comment]
Migration is done as per [the guide](https://material.io/design/navigation/understanding-navigation.html#reverse-navigation). It looks more modern to me and has smooth transition animation. I'm tweaking colors, layouts, and fonts to make it look more consistant.

**Comment by xcffl** [comment]
BTW, it's based on https://github.com/wallabag/android-app/pull/803

**Comment by msfjarvis** [comment]
I wanted to pursue this after my current PR was merged, glad to see someone else has taken initiative already 👍

**Comment by xcffl** [comment]
I realize that some customizations require refactorization. The first problem is whether it is necessary to have a "full-screen mode", which hides the top app bar. There is [no such API in AndroidX library](https://developer.android.com/reference/androidx/appcompat/widget/Toolbar).

As scrolling effectively hides the app bar, I wounder if it's possible that we simply remove this feature.

**Comment by di72nn** [comment]
@silberzwiebel #114 any opinion on that?

**Comment by msfjarvis** [comment]
> I realize that some customizations require refactorization. The first problem is whether it is necessary to have a "full-screen mode", which hides the top app bar. There is [no such API in AndroidX library](https://developer.android.com/reference/androidx/appcompat/widget/Toolbar).
> 
> As scrolling effectively hides the app bar, I wounder if it's possible that we simply remove this feature.

You can simply switch to having an [AppBarLayout](https://developer.android.com/reference/com/google/android/material/appbar/AppBarLayout.html) with a nested Toolbar be the `supportActionBar` and a `.NoActionBar` parent theme. Since AppBarLayout is a view, you can call `View.setVisibility` on it to remedy this problem.

**Comment by silberzwiebel** [comment]
> > As scrolling effectively hides the app bar, I wounder if it's possible that we simply remove this feature.
> 
> @silberzwiebel #114 any opinion on that?

Go ahead and feel free to remove it. I'm not using it anyway, because I was missing the "favorite" button below the article (so I had to disable fullscreen every 4th article or so). Plus, my implementation wasn't fully functional. If there is any way to have a proper fullscreen, I'd be glad to use it (but I'd need a "favorite" button below the article, then ;) ).

**Comment by xcffl** [comment]
> > > As scrolling effectively hides the app bar, I wounder if it's possible that we simply remove this feature.
> > 
> > 
> > @silberzwiebel #114 any opinion on that?
> 
> Go ahead and feel free to remove it. I'm not using it anyway, because I was missing the "favorite" button below the article (so I had to disable fullscreen every 4th article or so). Plus, my implementation wasn't fully functional. If there is any way to have a proper fullscreen, I'd be glad to use it (but I'd need a "favorite" button below the article, then ;) ).

Not sure if I will fix it (it takes no effect currently). A Favorite button sounds a good idea and not hard to implement (although fullscreen mode may leaves no exit). I'm not working on that anyway.


And here are some screenshots for current process:
![Screenshot_20190823-195711_wallabag](https://user-images.githubusercontent.com/2216902/63591292-dc6a4480-c59d-11e9-9639-0dc9b83f4fdf.png)
![Screenshot_20190823-195609_wallabag](https://user-images.githubusercontent.com/2216902/63591293-dd02db00-c59d-11e9-8d2f-23c003ad586f.png)
![Screenshot_20190823-195605_wallabag](https://user-images.githubusercontent.com/2216902/63591294-dd02db00-c59d-11e9-8c91-da0db6430fec.png)

Still needs a lot of tweaking to make nothing broken.

**Comment by msfjarvis** [comment]
Looking great! Is there a WIP branch somewhere on GitHub that I can use? I'd love to help speed this along.

**Comment by xcffl** [comment]
> Looking great! Is there a WIP branch somewhere on GitHub that I can use? I'd love to help speed this along.

Here is a nearly usable version: https://github.com/xcffl/wallabag-android/tree/xcffl/ui-refresh

Broken parts:
- Settings: haven't implement sub-fragments that are recommended by the new standard
- Connection Wizard: the content of the second step disappears...
- Article: I'm trying to implement a collapsing tool bar layout such that long titles are wrapped as described in MD Guidline
  [](https://material.io/components/assets/1id_xzCgqZtgdjZUENjw15zhz6qPbOjNL/topappbars-title-long.png)

I'm also trying to make the lists look like this:
[](https://material.io/components/assets/1wOwJGIeiALlR1EP4Hm97O3nPzha1f2VZ/topappbars-behavior-scrolling-sameelevation.mp4)

**Comment by xcffl** [comment]
I find that https://material.io/resources/build-a-material-theme/#how-to-using-android-studio gives a good template to build up a material theme.

**Comment by xcffl** [comment]
Most problems are fixed.

In terms of the current Settings implementation being deprecated, it's not supported by MD. So I'm refactoring it. Hope it won't have many conflicts against others' PR.
https://developer.android.com/guide/topics/ui/settings/
